test/metabase/upload_test.clj

```diff
@@ -114,17 +117,20 @@
            ["₿42.243646" 42.243646      float-type ".,"]
            ["-99.99¢"    -99.99         float-type ".,"]
            ["."          "."            vchar-type]
+           ;; String-related
            [(apply str (repeat 255 "x")) (apply str (repeat 255 "x")) vchar-type]
            [(apply str (repeat 256 "x")) (apply str (repeat 256 "x")) text-type]
            ["86 is my favorite number"   "86 is my favorite number"   vchar-type]
            ["My favorite number is 86"   "My favorite number is 86"   vchar-type]
-           ["2022-01-01"                    #t "2022-01-01"       date-type]
-           ["2022-01-01T01:00:00"           #t "2022-01-01T01:00" datetime-type]
-           ["2022-01-01T01:00:00.00"        #t "2022-01-01T01:00" datetime-type]
-           ["2022-01-01T01:00:00.000000000" #t "2022-01-01T01:00" datetime-type]]]
+           ;; Date-related
+           [" 2022-01-01 "                    #t "2022-01-01"             date-type]
+           [" 2022-01-01T01:00:00 "           #t "2022-01-01T01:00"       datetime-type]
+           [" 2022-01-01T01:00:00.00 "        #t "2022-01-01T01:00"       datetime-type]
+           [" 2022-01-01T01:00:00.000000000 " #t "2022-01-01T01:00"       datetime-type]
+           [" 2022-01-01T01:00:00.00-07:00 "  #t "2022-01-01T01:00-07:00" offset-dt-type]]]
```

Comment:
The implementation of `parse-offset-datetime` uses `t/offset-date-time`, which can tolerate more formats e.g.

```
(t/offset-date-time "2020-01-01T00:00Z")
;; => #t "2020-01-01T00:00Z"

(t/offset-date-time "2020-01-01T00:00+01")
;; => #t "2020-01-01T00:00+01:00"
```

I think we should tolerate these other formats too, but these tests don't verify that they are supported. I think we should add tests for them to be sure they are tolerated in the future.